src/test/scala/extractors/ExtractorSpec.scala

```diff
@@ -0,0 +1,115 @@
+package extractors
+
+import org.scalatest.GivenWhenThen
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.io.ByteArrayOutputStream
+
+class ExtractorSpec extends AnyFlatSpec with Matchers with GivenWhenThen {
+
+  "Email.apply function" should "take local part 'test' and domain part 'intellias.com' string as arguments and return email string 'test@intellias.com'" in {
+    Given("local part string 'test' and domain part string 'intellias.com'")
+    val localPart: String = "test"
+    val domain: String = "intellias.com"
+
+    When("apply function is called for Email object and given parameters")
+    val result = Email.apply(localPart, domain)
```

Comment:
If the object in scala has a method `apply`, it can be treated as a function, so instead of writing `Email.apply(localPart, domain)`, you can write: `Email(localPart, domain)`

Reply:
Changed to Email(localPart, domain)